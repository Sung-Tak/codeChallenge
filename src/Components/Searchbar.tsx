import React, {useState} from "react"


type IValues = {
    setValue: React.Dispatch<React.SetStateAction<string>>
}

const Searchbar = ({setValue}:IValues) =>{
    const [search, setSearch] = useState<string>("")

    const handleChange = (e:React.ChangeEvent<HTMLInputElement>) => {
        //console.log(e.currentTarget.value)
        setSearch(e.currentTarget.value)
        setValue(e.currentTarget.value.toLowerCase())
    }

    return <input 
    style={{position:"fixed", height:"2em", width:"400px", fontSize: "1.5em", left:"40%"}} 
    type="text" onChange={handleChange} 
    value={search}
    placeholder="Search By Title..."
    ></input>

}
export default Searchbar