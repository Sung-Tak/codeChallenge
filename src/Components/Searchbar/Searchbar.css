.searchbar{
    position: fixed;
    height: 2em;
    width: 96%;
    left:2%;
    font-size: 1.5em;
    z-index: 1;
}