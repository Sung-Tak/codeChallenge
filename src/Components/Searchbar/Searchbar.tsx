import React, { FC, useState } from "react"
import "./Searchbar.css"
type IValues = {
  setValue: React.Dispatch<React.SetStateAction<string>>
};

const Searchbar: FC<IValues> = ({ setValue }) => {
  const [search, setSearch] = useState<string>("")

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    //console.log(e.currentTarget.value)
    setSearch(e.currentTarget.value)
    setValue(e.currentTarget.value.toLowerCase())
  };

  return (
    <input
      className="searchbar"
      type="text"
      onChange={handleChange}
      value={search}
      placeholder="Search By Title..."
    ></input>
  )
}
export default Searchbar
