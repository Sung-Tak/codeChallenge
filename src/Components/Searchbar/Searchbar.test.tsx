import React, { useState } from "react";
import { render, screen } from "@testing-library/react";
import Searchbar from "./Searchbar";

describe("searchbar", () => {
  test("snapshot test", () => {
    const { asFragment } = render(<Searchbar setValue={() => {}} />);

    expect(asFragment()).toMatchSnapshot();
  });
  test("should render input in searchbar", () => {
    render(<Searchbar setValue={() => {}} />);
    const inputEl = screen.queryByPlaceholderText("Search By Title...");
    expect(inputEl).toBeInTheDocument();
  });
});
