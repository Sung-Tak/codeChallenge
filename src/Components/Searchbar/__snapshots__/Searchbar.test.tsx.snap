// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchbar snapshot test 1`] = `
<DocumentFragment>
  <input
    placeholder="Search By Title..."
    style="position: fixed; height: 2em; width: 400px; font-size: 1.5em; left: 40%;"
    type="text"
    value=""
  />
</DocumentFragment>
`;
